.github/workflows/reusable_test_suite.yml

```diff
@@ -39,41 +31,14 @@ jobs:
         id: checkout
         uses: actions/checkout@v4
 
-      - name: 'Determine files differing from target branch'
-        id: changed-files
-        if: ${{ github.event_name != 'push' && github.ref != 'refs/heads/main' }}
-        uses: tj-actions/changed-files@v44
-        with:
-          files: |
-            .github/workflows/test_suite.yml
-            ${{ inputs.changed-files-patterns }}
-
-      - name: 'Cleanup'
-        if: ${{ always() }}
-        run: |
-          cd ..
-          rm -rf build
-
-      - name: 'Setup Python'
-        if: ${{ (github.event_name == 'push' && github.ref == 'refs/heads/main') || (steps.changed-files.outcome == 'success' && steps.changed-files.outputs.any_changed == 'true') || github.event_name == 'schedule' }}
-        uses: actions/setup-python@v2
-        with:
-          python-version: '3.12'
-
-      - name: 'Install Package'
-        if: ${{ (github.event_name == 'push' && github.ref == 'refs/heads/main') || (steps.changed-files.outcome == 'success' && steps.changed-files.outputs.any_changed == 'true') || github.event_name == 'schedule' }}
-        run: |
-          . /home/firedrake/firedrake/bin/activate
-          ${{ inputs.install-command }}
-
       - name: 'Lint'
-        if: ${{ (github.event_name == 'push' && github.ref == 'refs/heads/main') || (steps.changed-files.outcome == 'success' && steps.changed-files.outputs.any_changed == 'true') || github.event_name == 'schedule' }}
         run: |
           . /home/firedrake/firedrake/bin/activate
           make lint
 
-      - name: 'Run Tests'
-        if: ${{ (github.event_name == 'push' && github.ref == 'refs/heads/main') || (steps.changed-files.outcome == 'success' && steps.changed-files.outputs.any_changed == 'true') || github.event_name == 'schedule' }}
+      - name: 'Run tests'
         run: |
           . /home/firedrake/firedrake/bin/activate
           ${{ inputs.test-command }}
```

Comment:
I left this in to allow setting the animate checkpoint directory in Animate's workflow